caffe2/operators/utility_ops.h

```diff
@@ -791,27 +791,33 @@ class ScatterOp : public Operator<CPUContext> {
     const IndexType* idxs = indices.template data<IndexType>();
     const char* src_base = static_cast<const char*>(updates.raw_data());
 
-    const int64_t outer_dims_product = updates.size_to_dim(axis_);
-    const int64_t block_size = updates.size_from_dim(axis_ + 1);
-    const int64_t block_bytesize = block_size * item_bytesize;
+    const int64_t outer_dims_product = indices.size_to_dim(axis_);
 
-    const int64_t src_indexing_axis_dim = updates.size(axis_);
-    const int64_t src_batch_bytesize = updates.size_from_dim(axis_) * item_bytesize;
-    const int64_t dst_batch_size = data.size_from_dim(axis_) * item_bytesize;
+    const int64_t dst_indexing_axis_dim = data.size(axis_);
+
+    const int64_t idxs_block_size = indices.size_from_dim(axis_ + 1);
+    const int64_t src_block_size = updates.size_from_dim(axis_ + 1);
+    const int64_t dst_block_size = data.size_from_dim(axis_ + 1);
+
+    const int64_t idxs_batch_size = indices.size_from_dim(axis_);
+    const int64_t src_batch_size = updates.size_from_dim(axis_);
+    const int64_t dst_batch_size = data.size_from_dim(axis_);
 
     const int64_t N = indices.size(axis_);
 
-    check_indexarray_range<IndexType>(idxs, N, src_indexing_axis_dim);
+    check_indexarray_range<IndexType>(idxs, N, dst_indexing_axis_dim);
 
-    int64_t i = 0;
-    for (int64_t batch = 0; batch < outer_dims_product; ++batch) {
-      int64_t i_max = i + N;
-      for (; i < i_max && i < indices.numel(); ++i) {
-        auto idx = idxs[i];
+    for (int64_t outer_batch = 0; outer_batch < outer_dims_product; ++outer_batch) {
```

Comment:
Add some comments about how we compute the index for source and destination, otherwise it's really hard to follow.

Reply:
agreed... adding comments to help clarification